File: semester 4/MAP/JFTS/FestivalDomain/src/main/java/ro/mpp/domain/Ticket.java
package ro.mpp.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Timestamp;

@Getter
@AllArgsConstructor
public class Ticket extends Entity<Integer> {
    private final String buyerName;
    @Setter
    private Integer numberOfSeats;
    private final Timestamp purchaseDate;
    private final Show show;

    public Ticket(ResultSet rs, Show show) throws SQLException {
        super(show.getId());
        this.numberOfSeats = rs.getInt("numberOfSeats");
        this.buyerName = rs.getString("buyerName");
        this.purchaseDate = rs.getTimestamp("purchaseDate");
        this.show = show;
    }
}

File: semester 4/MPP/C#FTS/Model/Domain/Ticket.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;

namespace C_FTS.Domain
{
    public class Ticket : Entity<int>
    {
        public string BuyerName { get; }
        public int NumberOfSeats { get; set; }
        public string PurchaseDate { get; }
        public Show Show { get; }

        public Ticket(int id, string buyerName, int numberOfSeats, string purchaseDate, Show show) : base(id)
        {
            BuyerName = buyerName;
            NumberOfSeats = numberOfSeats;
            PurchaseDate = purchaseDate;
            Show = show;
        }

        public Ticket(string buyerName, int numberOfSeats, string purchaseDate, Show show) : base(-1)
        {
            BuyerName = buyerName;
            NumberOfSeats = numberOfSeats;
            PurchaseDate = purchaseDate;
            Show = show;
        }

        public Ticket(IDataReader reader, Show show)
            : base(reader.GetInt32(reader.GetOrdinal("id")))
        {
            BuyerName = reader.GetString(reader.GetOrdinal("buyer_name"));
            NumberOfSeats = reader.GetInt32(reader.GetOrdinal("number_of_seats"));
            PurchaseDate = reader.GetString(reader.GetOrdinal("purchase_date"));
            Show = show;
        }

        public override string ToString()
        {
            return $"Ticket(id={Id}, BuyerName={BuyerName}, PurchesDate={PurchaseDate}, Show={Show})";
        }
    }
}
